pkg/api/provision.go

```diff
@@ -411,13 +439,65 @@ func (s *server) provision(w http.ResponseWriter, r *http.Request) {
 		s.writeResponse(w, http.StatusInternalServerError, responseEmptyJSON)
 		return
 	}
-
 	// If we get all the way to here, we've been successful!
 	s.writeResponse(w, http.StatusAccepted, responseProvisioningAccepted)
 
 	log.WithFields(logFields).Debug("asynchronous provisioning initiated")
 }
 
+func (s *server) isParentProvisioning(instance service.Instance) (bool, error) {
+	//No parent, so no need to wait
+	if instance.ParentAlias == "" {
+		return false, nil
+	}
+
+	parent, parentFound, err := s.store.GetInstanceByAlias(instance.ParentAlias)
+
+	//Parent has was not found, so wait for that that to occur
+	if !parentFound {
+		return true, nil
+	}
+
+	if err != nil {
+		log.WithFields(log.Fields{
+			"error":      "waitforParent",
+			"instanceID": instance.InstanceID,
+		}).Error(
+			"bad provision request: unable to retrieve parent",
+		)
+		return false, err
+	}
+
+	//If parent failed, we should not even attempt to provision this
+	if parent.Status == service.InstanceStateProvisioningFailed {
+		log.WithFields(log.Fields{
+			"error":      "waitforParent",
+			"instanceID": instance.InstanceID,
+		}).Error(
+			"bad provision request: parent failed provisioning",
+		)
+		return false, fmt.Errorf("error provisioning: parent provision failed")
```

Comment:
imo, this should be severity `INFO`; not `ERROR`. My reasoning (which I have applied as consistently as I've been able throughout the rest of the broker) is that we use `ERROR` when the broker itself has experienced a problem. In this case, it has not. The broker refusing to provision a child for a parent that has already failed provisioning is actually the broker working as designed. `INFO` because is _is_ worth knowing about, but not `ERROR` because this isn't evidence that anything is wrong with the broker.